src/steps/big-table/converters.ts

```diff
@@ -0,0 +1,222 @@
+export function getInstanceKey(instance: bigtableadmin_v2.Schema$Instance) {
+  return `bigtable_instance:${instance.name}`;
+}
+
+export function getAppProfileKey(
+  appProfile: bigtableadmin_v2.Schema$AppProfile,
+) {
+  return `bigtable_appProfile:${appProfile.name}`;
+}
+
+export function getClusterKey(cluster: bigtableadmin_v2.Schema$Cluster) {
+  return `bigtable_cluster:${cluster.name}`;
+}
+
+export function getBackupKey(backup: bigtableadmin_v2.Schema$Backup) {
+  return `bigtable_backup:${backup.name}`;
+}
+
+export function getTableKey(table: bigtableadmin_v2.Schema$Table) {
+  return `bigtable_table:${table.name}`;
+}
+
+export function getLocationKey(location: bigtableadmin_v2.Schema$Location) {
+  return `bigtable_location:${location.locationId}`;
+}
+
+export function createOperationEntity(
+  operation: bigtableadmin_v2.Schema$Operation,
+) {
+  return createGoogleCloudIntegrationEntity(operation, {
+    entityData: {
+      source: operation,
+      assign: {
+        _class: ENTITY_CLASS_BIG_TABLE_OPERATION,
+        _type: ENTITY_TYPE_BIG_TABLE_OPERATION,
+        _key: getOperationKey(operation),
+        name: operation.name,
+        done: operation.done,
+      },
+    },
+  });
+}
+
+export function createInstanceEntity({
+  instance,
+  projectId,
+}: {
+  instance: bigtableadmin_v2.Schema$Instance;
+  projectId: string | undefined | null;
+}) {
+  const instanceName = getLastUrlPart(instance.name!);
+
+  return createGoogleCloudIntegrationEntity(instance, {
+    entityData: {
+      source: instance,
+      assign: {
+        _class: ENTITY_CLASS_BIG_TABLE_INSTANCE,
+        _type: ENTITY_TYPE_BIG_TABLE_INSTANCE,
+        _key: getInstanceKey(instance),
+        name: instance.name,
+        displayName: instance.displayName!,
+        active: instance.state === 'READY',
+        state: instance.state,
+        type: instance.type,
+        webLink: getGoogleCloudConsoleWebLink(
+          `/bigtable/instances/${instanceName}/overview?project=${projectId}`,
+        ),
+      },
+    },
+  });
+}
+
+export function createAppProfileEntity({
+  appProfile,
+  projectId,
+  instanceId,
+}: {
+  appProfile: bigtableadmin_v2.Schema$AppProfile;
+  projectId: string | undefined | null;
+  instanceId: string | undefined | null;
+}) {
+  const instanceName = getLastUrlPart(instanceId!);
+  const appProfileName = getLastUrlPart(appProfile.name!);
+
+  return createGoogleCloudIntegrationEntity(appProfile, {
+    entityData: {
+      source: appProfile,
+      assign: {
+        _class: ENTITY_CLASS_BIG_TABLE_APP_PROFILE,
+        _type: ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
+        _key: getAppProfileKey(appProfile),
+        name: appProfile.name,
+        instanceId,
+        etag: appProfile.etag,
+        description: appProfile.description,
+        webLink: getGoogleCloudConsoleWebLink(
+          `/bigtable/instances/${instanceName}/app-profiles/${appProfileName}?project=${projectId}`,
+        ),
+      },
+    },
+  });
+}
```

Comment:
Can you add the following properties (also update the tests with them):

```
"multiClusterRoutingUseAny.done": appProfile.multiClusterRoutingUseAny.done
"multiClusterRoutingUseAny.error": appProfile.multiClusterRoutingUseAny.error.message
"multiClusterRoutingUseAny.errorCode": appProfile.multiClusterRoutingUseAny.error.code
"multiClusterRoutingUseAny.name": appProfile.multiClusterRoutingUseAny.name
"singleClusterRouting.allowTransactionalWrites": appProfile.singleClusterRouting.allowTransactionalWrites
"singleClusterRouting.clusterId": appProfile.singleClusterRouting.clusterId
```

In general, if there are some seemingly useful fields hidden behind multiple nested objects we can use this "dot" notation to separate them (if naming is otherwise hard). Not ideal, but could work.

Reply:
<img width="664" alt="image" src="https://user-images.githubusercontent.com/39617427/129123254-770252a4-eb07-42a8-ab41-2ef5575f418f.png">

Typescript is complaining that these properties don't exist on the schema provided by J1